types/atlassian-crowd-client/index.d.ts

```diff
@@ -0,0 +1,116 @@
+// Type definitions for atlassian-crowd-client 2.0
+// Project: https://github.com/ghengeveld/atlassian-crowd-client
+// Definitions by: mtgto <https://github.com/mtgto>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import CrowdApi from "./api";
+import { Settings } from "./settings";
+import Attributes from "./lib/models/attributes";
+import Group from "./lib/models/group";
+import User from "./lib/models/user";
+import Session from "./lib/models/session";
+import ValidationFactors from "./lib/models/validation-factors";
+
+export default class CrowdClient extends CrowdApi {
```

Comment:
the package at runtime does not have `default` properties.

this shoudl be `export = CrowdClient` instead of `export default CrowdClient`

```shell
c:\test\dt>npm install atlassian-crowd-client
+ atlassian-crowd-client@2.0.0
added 2 packages from 1 contributor and audited 5 packages in 1.384s
found 0 vulnerabilities

c:\test\dt>node
> var a = require("atlassian-crowd-client");
undefined
> a;
[Function: CrowdClient]
```

See https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default for more details.

Reply:
This also applies to all the other modules

Reply:
Ok, I'll fix. But I have a question.

`atlassian-crowd-client` is written in ES6. It uses `export default` not `export =`.
example:
https://github.com/ghengeveld/atlassian-crowd-client/blob/v2.0.0/src/client.js#L9

The link says
> For an NPM package, export = is accurate if node -p 'require("foo")' is the export, and export default is accurate if node -p 'require("foo").default' is the export.

I think it is prefer to use `export default` in type definition because the package use too.
But I tested, and find you are right:
```
$ node
> var CrowdClient = require('atlassian-crowd-client').default;
undefined
> new CrowdClient();
TypeError: CrowdClient is not a constructor
```

I don't understand why node can't understand `export default`...

Reply:
I understand, perhaps.

client.js in npm package doesn't have `export default`.
the package is compiled with babel.

↓The last of `lib/client.js` in npm package.

```javascript
...
exports['default'] = CrowdClient;
module.exports = exports['default'];
```

Reply:
@mhegazy I somehow fixed it.
(Do I need to mention you after fixed? I'm not know.)